dali/kernels/imgproc/convolution/convolution_gpu_test.cu

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <cmath>
+#include <complex>
+#include <tuple>
+#include <vector>
+
+#include "dali/core/boundary.h"
+#include "dali/core/convert.h"
+#include "dali/kernels/common/utils.h"
+#include "dali/kernels/imgproc/convolution/baseline_convolution.h"
+#include "dali/kernels/imgproc/convolution/convolution_cpu.h"
+#include "dali/kernels/imgproc/convolution/convolution_gpu.h"
+#include "dali/kernels/scratch.h"
+#include "dali/test/tensor_test_utils.h"
+#include "dali/test/test_tensors.h"
+
+namespace dali {
+namespace kernels {
+
+template <int ndim_, bool has_channels_, int axis_, int window_size_, typename InType_>
+struct convolution_params {
+  static constexpr int ndim = ndim_;
```

Comment:
I think we tend to use `_` for member fields than for the arguments.

Reply:
For classes, this is struct and for structs we don't need `_` for members.